playback/src/player.rs

```diff
@@ -25,48 +25,121 @@ use crate::audio_backend::Sink;
 use crate::metadata::{AudioItem, FileFormat};
 use crate::mixer::AudioFilter;
 
+const PRELOAD_NEXT_TRACK_BEFORE_END_DURATION_MS: u32 = 30000;
```

Comment:
Do we think this should be configurable through a CLI flag? Might be too little for slow networks.

Reply:
30 seconds should be enough to load a track, especially with the new loading code I wrote last year. If you can't start playback of a track within 30 seconds you won't have much fun with streaming music anyway...

I chose 30 seconds because in past discussions I read it was mentioned that the official client starts to load the next track 30 seconds before the end.

Loading too early can potentially hurt in two ways:
1) More wasted bandwidth due to a higher chance of the user not actually playing the next track.
2)
- The user starts a track and quickly seeks towards the end.
- The file loader has loaded everything from the current play position to the end of the track and the pre-load is triggered.
- The user seeks back towards the beginning of the track which hasn't yet been loaded in full.
- The pre-load is still using bandwidth while the current song must be downloaded as well.
- If the user doesn't have enough bandwidth for two simultaneous downloads, there will be buffer-underruns.

If you really feel it should be a config option, I can implement it.

Reply:
I believe 30 seconds is the default in the Spotify client. The points you make are fair enough, and thinking about it a flag for preload time is probably a solution in search of a problem. If someone has problems, we can always implement it at a later date.